### PR TITLE
feat: permissionless AMM pools from Interchain assets

### PR DESCRIPTION
**Pull request by dckc**
refs: #5075, #4643

## Description

Given sufficient IST, create an issuer for an IBC denom and an invitation to add a pool for it to the AMM.

There are 2 steps:
  1. offer IST (Central) and provide a denom in the offer args; get back the issuer for that denom in the VBANK
  2. offer the new asset (Secondary) and get back liquidity.

Un-ties knots discussed in https://github.com/Agoric/agoric-sdk/issues/4643#issuecomment-1130411097 , https://github.com/Agoric/agoric-sdk/issues/5075#issuecomment-1115225570

TODO:
 - [x] `bankManager` should have a method like `addAsset` that does _not_ propagate to every wallet in the system.
   - postponed as #5412
 - [x] bootstrap wiring

### Security Considerations

Does not provide offer safety, due to usual propoerties of `offerTo`.

Requires `bankManager` as a private arg.

### Documentation Considerations

Yet Another Contract to document / review.
cc @jessysaurusrex 

### Testing Considerations

Lightly tested. Reviewers will please suggest further tests.

### CODE REVIEW

File: packages/run-protocol/src/interchainPool.js
Comment (dckc) [inline packages/run-protocol/src/interchainPool.js]: @Chris-Hibbert the `addPool` call would move to step 2 when we require funds (#5377). I can't move it yet because the AMM requires the liquidity brand in the **want** when adding liquidity, even though it doesn't use it.

Maybe I should fix that in this PR?

Comment (Chris-Hibbert) [reply packages/run-protocol/src/interchainPool.js]: > @Chris-Hibbert the addPool call would move to step 2 when we require funds (https://github.com/Agoric/agoric-sdk/pull/5377). I can't move it yet because the AMM requires the liquidity brand in the want when adding liquidity, even though it doesn't use it.

> Maybe I should fix that in this PR?

Before and after #5377 the APIs for adding a pool are just different. Our choices are to reconcile in whichever PR is second, or merge this into #5377 once approved.

File: packages/run-protocol/src/interchainPool.js
Comment (Chris-Hibbert) [inline packages/run-protocol/src/interchainPool.js:21]: If this should eventually be governed, let's at least add a comment to that effect here.

File: packages/run-protocol/src/interchainPool.js
Comment (Chris-Hibbert) [inline packages/run-protocol/src/interchainPool.js:108]: I think there should be a `.catch()` clause here that returns the money from seat2 to seat1 and exits the seat.

I would test it by setting the AMM's minimum higher than the minimum enforced here.

**Review by Chris-Hibbert — APPROVED** [review]

File: packages/run-protocol/src/interchainPool.js
Comment (dckc) [reply packages/run-protocol/src/interchainPool.js:21]: I added a TODO to get this value dynamically from the AMM (once #5377 lands)

File: packages/run-protocol/src/interchainPool.js
Comment (dckc) [reply packages/run-protocol/src/interchainPool.js:108]: I plan to get the AMM's minimum on demand, so that test wouldn't last.

What is it that could throw? `seat.exit()` or `deposited`? In either case, what difference does it make which seat the money is returned in?

File: packages/run-protocol/src/interchainPool.js
Comment (dckc) [reply packages/run-protocol/src/interchainPool.js]: I did fix the odd AMM constraint and move `addPool` to step 2. That way, when #5377 lands, we'll have the collateral on hand when we want to `addPool`.

**Comment by dckc** [comment]
@Chris-Hibbert @turadg I wired up everything to get `test-gov-collateral.js` and such working. Please take a look.

File: packages/run-protocol/src/interchainPool.js
Comment (turadg) [inline packages/run-protocol/src/interchainPool.js]: what does `ISSUE` mean? something to do or just know about?

File: packages/run-protocol/test/test-gov-collateral.js
Comment (turadg) [inline packages/run-protocol/test/test-gov-collateral.js:156]: `@ts-expect-error`

(not worth a push bc I plan to lint this soon)

**Review by turadg — COMMENTED** [review]
Nothing stands out to me as a concern but given the security risks and my understanding of the interchain I don't feel comfortable approving and leave that to @Chris-Hibbert .

File: packages/run-protocol/src/interchainPool.js
Comment (dckc) [reply packages/run-protocol/src/interchainPool.js]: > what does `ISSUE` mean?

um... something to do, I suppose.

In this case, it originated from discussion with @dtribble , who was OK postponing it; I didn't turn it into a separate issue, #5412 , until I got a chance to run it by @michaelfig , where we decided it's feasible in due course. Now that I have a number for it, I can add it to that comment.

File: packages/run-protocol/src/interchainPool.js
Comment (Chris-Hibbert) [reply packages/run-protocol/src/interchainPool.js:108]: > what difference does it make which seat the money is returned in?

I thought seat2 was purely internal, but I now see that I was wrong.

It still seems surprising that the caller has to check both seats to see if there's money there.  

If someone has already saved an issuer on the AMM and called it 'Interchain5', then the `saveIssuer()` call will fail, and the allocation will be stuck on seat2.

**Review by Chris-Hibbert — APPROVED** [review]

File: packages/run-protocol/src/interchainPool.js
Comment (dckc) [reply packages/run-protocol/src/interchainPool.js:108]: > It still seems surprising that the caller has to check both seats to see if there's money there.

In discussion with @dtribble and co., we didn't see an obvious improvement. (There are 2 offers, so having 2 seats to check on failure never seemed odd, to me.)

> If someone has already saved an issuer on the AMM and called it 'Interchain5', then the `saveIssuer()` call will fail

The `saveIssuer()` call here doesn't depend on issuers in the AMM, as we discussed. (There could be a collision at `addPool`, but that's before the `saveIssuer()` here.)

**Pull request merged**
